src/pages/resultsView/pathwayMapper/PathwayMapperTable.tsx

```diff
@@ -0,0 +1,142 @@
+
+import * as React from 'react';
+import * as _ from "lodash";
+import { observer } from "mobx-react";
+import LazyMobXTable, { Column } from "shared/components/lazyMobXTable/LazyMobXTable";
+import { observable } from 'mobx';
+import { Radio } from 'react-bootstrap';
+import DefaultTooltip from 'public-lib/components/defaultTooltip/DefaultTooltip';
+
+export interface IPathwayMapperTable {
+    name: string;
+    score: number;
+    genes: string[];
+}
+
+enum IPathwayMapperTableColumnType {
+    NAME,
+    SCORE,
+    GENES
+}
+
+interface IPathwayMapperTableProps {
+    data: IPathwayMapperTable[];
+    selectedPathway: string;
+    changePathway: Function;
+    initialSortColumn?: string;
+}
+
+type PathwayMapperTableColumn = Column<IPathwayMapperTable>;
+
+class PathwayMapperTableComponent extends LazyMobXTable<IPathwayMapperTable> {
+}
+
+@observer
+export default class PathwayMapperTable extends React.Component<IPathwayMapperTableProps> {
+    public static defaultProps = {
+        columns: [
+            IPathwayMapperTableColumnType.NAME,
+            IPathwayMapperTableColumnType.SCORE,
+            IPathwayMapperTableColumnType.GENES,
+        ],
+        initialSortColumn: "name"
+    };
+    @observable protected _columns: { [columnEnum: number]: PathwayMapperTableColumn };
+    @observable selectedPathway: string = "";
+
+    constructor(props: IPathwayMapperTableProps) {
+        super(props);
+        this._columns = {};
+        this.generateColumns();
+    }
+
+    generateColumns() {
```

Comment:
minor but in general i think it's better to have a function return something which the calling code can assign to an instance property rather than a method which operates on an instance property (e.g. this._columns).